Sources/Tracking/CustomerIO.swift

```diff
@@ -21,339 +21,55 @@ public extension CustomerIOInstance {
     }
 }
 
-/**
- Welcome to the Customer.io iOS SDK!
-
- This class is where you begin to use the SDK.
- You must call `CustomerIO.initialize` to use the features of the SDK.
- */
-public class CustomerIO: CustomerIOInstance {
-    /// The current version of the Customer.io SDK.
-    public static var version: String {
-        SdkVersion.version
-    }
-
-    public var siteId: String? {
-        diGraph?.sdkConfig.siteId
-    }
-
-    /**
-     Singleton shared instance of `CustomerIO`. Convenient way to use the SDK.
-
-     Note: Don't forget to call `CustomerIO.initialize()` before using this!
-     */
-    @Atomic public private(set) static var shared = CustomerIO()
-
-    // Only assign a value to this *when the SDK is initialzied*.
-    // It's assumed that if this instance is not-nil, the SDK has been initialized.
-    // Tip: Use `SdkInitializedUtil` in modules to see if the SDK has been initialized and get data it needs.
-    private var implementation: CustomerIOInstance?
-
-    // The 1 place that DiGraph is strongly stored in memory for the SDK.
-    // Exposed for `SdkInitializedUtil`. Not recommended to use this property directly.
-    var diGraph: DIGraph?
-
-    // strong reference to repository to prevent garbage collection as it runs tasks in async.
-    @Atomic private var cleanupRepository: CleanupRepository?
-
-    // private constructor to force use of singleton API
-    private init() {}
-
-    // Constructor for unit testing. Just for overriding dependencies and not running logic.
-    // See CustomerIO.shared.initializeIntegrationTests for integration testing
-    init(implementation: CustomerIOInstance, diGraph: DIGraph) {
-        self.implementation = implementation
-        self.diGraph = diGraph
-    }
-
-    /**
-     Make testing the singleton `instance` possible.
-     Note: It's recommended to delete app data before doing this to prevent loading persisted credentials
-     */
-    static func resetSharedInstance() {
-        shared = CustomerIO()
-    }
-
-    // Special initialize used for integration tests. Mostly to be able to share a DI graph
-    // between the SDK classes and test class. Runs all the same logic that the production `intialize` does.
-    static func initializeIntegrationTests(
-        diGraph: DIGraph
-    ) {
-        let implementation = CustomerIOImplementation(diGraph: diGraph)
-        shared = CustomerIO(implementation: implementation, diGraph: diGraph)
-        shared.postInitialize(diGraph: diGraph)
-    }
-
-    /**
-     Initialize the shared `instance` of `CustomerIO`.
-     Call this function when your app launches, before using `CustomerIO.instance`.
-     */
-    @available(iOSApplicationExtension, unavailable)
-    public static func initialize(
+public extension CustomerIO {
+    // Initialize for general usage
+    static func initialize(
```

Comment:
I think this function needs `@available(iOSApplicationExtension, unavailable)` added to the top. 

That may fix [this](https://github.com/customerio/customerio-ios/pull/411/files#r1400703149).